scripts/ci-entrypoint.sh

```diff
@@ -194,7 +215,11 @@ if [[ -n "${TEST_CCM:-}" ]]; then
 --set cloudNodeManager.imageName="${CNM_IMAGE_NAME}" \
 --set-string cloudControllerManager.imageTag="${IMAGE_TAG}" \
 --set-string cloudNodeManager.imageTag="${IMAGE_TAG}" \
---set cloudControllerManager.replicas="${CCM_COUNT}"
+--set cloudControllerManager.replicas="${CCM_COUNT}" \
+--set cloudControllerManager.enableDynamicReloading="${ENABLE_DYNAMIC_RELOADING}"  \
+--set cloudControllerManager.cloudConfig="${CLOUD_CONFIG}" \
+--set cloudControllerManager.cloudConfigSecretName="${CONFIG_SECRET_NAME}"
```

Comment:
is this a new feature in cloud-provider? do you recommend to enable it outside e2e tests too (for end users)?

Reply:
> is this a new feature in cloud-provider? do you recommend to enable it outside e2e tests too (for end users)?

Actually it is an old feature https://cloud-provider-azure.sigs.k8s.io/install/configs/#dynamically-reloading-cloud-controller-manager, which is already exposed to end users. 
Currently, we need test cloud-config in cloud-provider-azure. It is much easier to change cloud config in the test code if CCM reads the config from the secret.

Reply:
Got it. I wonder if we should just do this always since it seems better than having the secret on the VM and also makes it easier to change it as you point out. @jackfrancis thoughts?

@monianshouhou do you know of any downsides of using a k8s secret for cloud config? Maybe ACR login won't work?

Reply:
> Got it. I wonder if we should just do this always since it seems better than having the secret on the VM and also makes it easier to change it as you point out. @jackfrancis thoughts?
> 
> @monianshouhou do you know of any downsides of using a k8s secret for cloud config? Maybe ACR login won't work?

Thanks for the reply! Sorry about that I don't know any downside of using the secret. However, reading from secret is a feature proposed 3 years ago and nowadays azure.json isn't deprecated. Maybe there exist some reasons. I suggest that we could give user freedom to choose load cloud-config either from azure.json or from secret, just like in cloud-provider-azure.
@lzhecheng What do you think?

Reply:
I'm pretty sure that the helm template layer uses sprig for the `hasKey` dict func:

- https://github.com/Masterminds/sprig/blob/master/dict.go#L25

This matters because the current cloud-provider-azure helm chart uses that to determine whether or not to include the `--cloud-config-secret-name` runtime configuration:

- https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/helm/cloud-provider-azure/templates/cloud-provider-azure.yaml#L199-L201

So, this change to capz's `ci-entrypoint.sh` will mean that we'll be invoking the `cloud-controller-manager` runtime in the following way when not using the secret:

```
cloud-controller-manager --cloud-config="/etc/kubernetes/azure.json" --cloud-config-secret-name=""...
```

compared to the way we do it now:

```
cloud-controller-manager --cloud-config="/etc/kubernetes/azure.json"...
```

Is this new change a supported, equivalent configuration for `cloud-controller-manager`? In other words, is it equivalent to pass in `--cloud-config-secret-name=""` (empty string) vs not passing in `--cloud-config-secret-name` configuration at all (this latter pattern is what we do right now)?

Reply:
> I'm pretty sure that the helm template layer uses sprig for the `hasKey` dict func:
> 
> * https://github.com/Masterminds/sprig/blob/master/dict.go#L25
> 
> This matters because the current cloud-provider-azure helm chart uses that to determine whether or not to include the `--cloud-config-secret-name` runtime configuration:
> 
> * https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/helm/cloud-provider-azure/templates/cloud-provider-azure.yaml#L199-L201
> 
> So, this change to capz's `ci-entrypoint.sh` will mean that we'll be invoking the `cloud-controller-manager` runtime in the following way when not using the secret:
> 
> ```
> cloud-controller-manager --cloud-config="/etc/kubernetes/azure.json" --cloud-config-secret-name=""...
> ```
> 
> compared to the way we do it now:
> 
> ```
> cloud-controller-manager --cloud-config="/etc/kubernetes/azure.json"...
> ```
> 
> Is this new change a supported, equivalent configuration for `cloud-controller-manager`? In other words, is it equivalent to pass in `--cloud-config-secret-name=""` (empty string) vs not passing in `--cloud-config-secret-name` configuration at all (this latter pattern is what we do right now)?

Thanks for the review! Yes, because if `cloud-config` is set, `cloud-config-secret-name` will be ignored.